### PR TITLE
Renovate/web

### PR DESCRIPTION
**Pull request by Xavier-Cliquennois**
### Problem

👋 Thanks for opening a [pull request](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#submitting-a-pull-request)! Please include a brief summary of the problem your change is trying to solve, or bug fix. If your change fixes a bug or you'd like to provide context on why you're making the change, please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) as follows:

Closes: None

### Solution

Please describe your change as it relates to the problem, or bug fix, as well as any dependencies. If your change requires a database schema migration, please describe the schema modification(s) and whether it's a _backwards-incompatible_ or _backwards-compatible_ change.

> **Note:** All database schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

One-line summary: The goal was to upgrade the web client to React 18 in order to utilize the latest version of Node.js and update all dependencies to their respective latest versions. Everything is good only the unit tests haven't been fully migrated to Jest and test-library for the moment.

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

**Comment by wslulciuc** [comment]
@Xavier-Cliquennois, thanks for helping out with the react `18` migration! I've added @phixMe as a reviewer to your PR.

File: web/webpack.common.js [hunk 1]
```diff
@@ -19,12 +19,12 @@ module.exports = {
...
-            options: {
-              plugins: [postCssModulesValues, autoprefixer],
-            },
-          }
+          // {
+          //   loader: 'postcss-loader',
+          //   options: {
+          //     plugins: [postCssModulesValues, autoprefixer],
+          //   },
+          // }
```
Comment (phixMe) [inline web/webpack.common.js:27]: If we are not using this, it should be deleted.

**Review by phixMe — COMMENTED** [review]
Hey, thanks for taking this large undertaking of a task.

One of the most important things is that the project builds with the following command out of the Readme
```sh
./docker/up.sh --build --seed
...
 > [web 8/9] RUN npm run build:
#0 0.470 
#0 0.470 > marquez-ui@1.0.0 build /usr/src/app
#0 0.470 > webpack --config webpack.prod.js
#0 0.470 
#0 0.722 [webpack-cli] Failed to load '/usr/src/app/webpack.prod.js' config

```
We're getting the following error with this PR.

File: web/src/__tests__/components/Lineage.test.tsx [hunk 2]
```diff
@@ -1,112 +0,0 @@
...
-      ['dataset_foo', 'job_foo'],
-      ['dataset_foo', 'job_foo'],
-      ['job_bar', 'dataset_foo'],
-      ['dataset_bar', 'job_bar'],
-      ['job_foo', 'dataset_bar']
-    ]
-
-    expect(actualPaths).toEqual(expectedPaths)
-  })
-})
```
[the remark this answers is not in the document]
Comment (Xavier-Cliquennois) [reply web/src/__tests__/components/Lineage.test.tsx]: I redo this test and add it in this [commit](https://github.com/MarquezProject/marquez/pull/2520/commits/669fbd5049c421f334bd430c3fba48ab34450c90)

File: web/db.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (Xavier-Cliquennois) [reply web/db.json]: Remove in this [commit](https://github.com/MarquezProject/marquez/pull/2520/commits/5ba3f9d0c7c411850a1c2ab556e80781944e70ca)

**Comment by Xavier-Cliquennois** [comment]
@phixMe I fixed the build with this [commit](url).
Regards

**Comment by Xavier-Cliquennois** [comment]
Hello @wslulciuc, is Slack functioning properly? I encountered a 404 error when accessing the link provided in the CONTRIBUTING.md file. I would like to discuss some ideas we have at [Graphite](https://wearegraphite.io/) for the frontend client with the team.

**Review by phixMe — COMMENTED** [review]
I found a couple small things. It looks like you've got some merge conflicts to resolve with some of the recent changes.

**Comment by wslulciuc** [comment]
@Xavier-Cliquennois, yikes! We recently updated our slack link and missed a couple references (see https://github.com/MarquezProject/marquez/pull/2532). We should be good now, but also, here's the link https://bit.ly/MqzSlack 😅

**Comment by Xavier-Cliquennois** [comment]
@wslulciuc Hello, 
I created an issue (#2533) about Slack connection.
Regards

File: web/src/components/core/small-icon/MqSmallIcon.tsx [hunk 4]
```diff
@@ -24,26 +13,28 @@ interface OwnProps {
...
-  classes
+  foregroundColor
 }) => {
   return (
     <Box
       width={16}
       height={16}
       bgcolor={backgroundColor}
-      borderRadius={shape === 'circle' ? '50%' : '4px'}
+      borderRadius="undefinedpx"
```
[the remark this answers is not in the document]
Comment (Xavier-Cliquennois) [reply web/src/components/core/small-icon/MqSmallIcon.tsx]: @phixMe True it's a mistake, thanks a lot, I just fix it in this [commit](https://github.com/MarquezProject/marquez/pull/2520/commits/56c917dee2e68a3854f23b701eac7712b5f5f13a)

File: web/src/__tests__/components/Lineage.test.tsx [hunk 5]
```diff
@@ -67,22 +79,23 @@ const mockGraphWithCycle = [
   }
 ]
 
-function mockSetState(newState: any) {
-  this.state = { ...this.state, ...newState }
-}
+// function mockSetState(newState: any) {
+//   this.state = { ...this.state, ...newState }
+// }
```
[the remark this answers is not in the document]
Comment (Xavier-Cliquennois) [reply web/src/__tests__/components/Lineage.test.tsx]: Done in this [commit](https://github.com/MarquezProject/marquez/pull/2520/commits/c82bda60db92bc6f957ba9bb55c24d4029826b2d)

**Comment by Xavier-Cliquennois** [comment]
@phixMe I have performed the conflict merge. 

The tests failed during the CI process before the tests were executed. However, all the tests passed locally.

https://github.com/MarquezProject/marquez/assets/6882044/735c670d-bc47-490e-b316-ae61aaa5ba9c

I hope there have been no new merges that modify the web code before this pull request is merged, as it could make the merge process difficult in my branch.

**Comment by wslulciuc** [comment]
> I hope there have been no new merges that modify the web code before this pull request is merged, as it could make the merge process difficult in my branch

@Xavier-Cliquennois we'll pause merging any noncritical PRs for web until we merge your sweet changes.

**Review by wslulciuc — APPROVED** [review]
Really amazing to see the UI getting some much needed love ❤️. Great work on the migration to node `18`, @Xavier-Cliquennois 💯 🥇

**Review by phixMe — CHANGES_REQUESTED** [review]
I'm looking through the app and I think several of the stylesheets have been changed due to the bump in material. I think some of the updates we are ok to inherit, but others are a little too disruptive. There are also a couple of regressions that we need to take care of before merging. I assume some of them are just inherited from upgrades.

Namespace should be centered with the text here. 
image

Hover states and group padding seems to have been removed here.
image

These buttons have a double border on the lineage events page.
image

Clicking the select state icon does not open the dropdown.
image
image

Link text should underline when hovered.
image

In general, the inputs have too much padding compared to before.
image

The pull down menu on the lineage page no longer works.
image

The "Zoom" on the lineage graph no longer works.
image

File: web/db.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (phixMe) [reply web/db.json:1]: Did this get added back in?

File: web/src/components/core/chip/MqChip.tsx [hunk 6]
```diff
@@ -21,41 +20,33 @@ interface MqChipProps {
...
   selectable
 }) => {
+
+  const theme = createTheme(useTheme())
+
   return (
     <Box
       id={`chip-${value}`}
-      className={classNames(classes.root, selected && classes.selected)}
+      sx={Object.assign({
```
Comment (phixMe) [inline web/src/components/core/chip/MqChip.tsx:42]: Do we need to do Object assign here instead of just directly applying the object?

Comment (Xavier-Cliquennois) [reply web/src/components/core/chip/MqChip.tsx]: Fix

File: web/db.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Xavier-Cliquennois) [reply web/db.json:1]: Yes, I used a json server and I forgot to remove the file before the commit.

**Comment by Xavier-Cliquennois** [comment]
@phixMe Hello, I did all the fixes you asked. Tell me if you found something else.

File: web/src/components/lineage/Lineage.tsx [hunk 7]
```diff
@@ -248,22 +249,23 @@ export class Lineage extends React.Component<LineageProps, LineageState> {
...
+                {zoom => (
+                  <div>
                       <svg
                         id={'GRAPH'}
                         width={parent.width}
                         height={parent.height}
                         style={{
                           cursor: zoom.isDragging ? 'grabbing' : 'grab'
                         }}
+                      // @ts-ignore
```
Comment (phixMe) [inline web/src/components/lineage/Lineage.tsx:263]: This makes our linter fail. I'd really prefer not to add this if possible.

File: web/src/components/lineage/components/drag-bar/DragBar.tsx [hunk 8]
```diff
@@ -4,100 +4,97 @@
...
-  }
+  // React.useEffect(() => {
+  //   window.addEventListener('mousemove', handleMousemove)
+  //   window.addEventListener('mouseup', handleMouseup)
+
+  //   return () => {
+  //     window.removeEventListener('mousemove', handleMousemove)
+  //     window.removeEventListener('mouseup', handleMouseup)
+  //   }
+  // }, [])
```
Comment (phixMe) [inline web/src/components/lineage/components/drag-bar/DragBar.tsx:36]: I think we do not want a comment like this lying around.

File: web/src/components/lineage/components/node/Node.tsx [hunk 9]
```diff
@@ -43,80 +43,77 @@ class Node extends React.Component<NodeProps> {
...
+  const ariaDatasetLabel = 'Dataset'
+  return (
+    <Link
+      to={determineLink(node)}
+      onClick={() => node.label && setSelectedNode(node.label)}
+    >
+      {job ? (
+        <g>
+          {/* { console.log(job.latestRun)} */}
+          {/* {console.log(runStateToNodeColor(job.latestRun))} */}
```
Comment (phixMe) [inline web/src/components/lineage/components/node/Node.tsx:58]: We can drop this, it shouldn't be here.

**Review by phixMe — COMMENTED** [review]
Nice work! We're getting really close!

I think this element looks misaligned and off center, it has some stylesheet padding that doesn't look right internally as well.

image

The language selector does not allow for clicking on the icon itself.
image

The drag bar selector no longer looks like a handle icon. Before it looked like this.
image
image

File: web/src/components/lineage/Lineage.tsx (hunk 7 above)
Comment (Xavier-Cliquennois) [reply web/src/components/lineage/Lineage.tsx]: Done

File: web/src/components/lineage/components/node/Node.tsx (hunk 9 above)
Comment (Xavier-Cliquennois) [reply web/src/components/lineage/components/node/Node.tsx]: Done

File: web/src/components/lineage/components/drag-bar/DragBar.tsx (hunk 8 above)
Comment (Xavier-Cliquennois) [reply web/src/components/lineage/components/drag-bar/DragBar.tsx]: Done

**Review by phixMe — APPROVED** [review]
Thanks @Xavier-Cliquennois I took a look at your remaining changes and they look great! Thanks for this massive contribution. It looks like you have some conflicts to resolve in the lock file, but otherwise let's merge this update in.

**Pull request closed without merging**